### ISSUE TITLE
[ not an issue ] Can the lightpad layout be used for showing many kinds of « recent activities » ?

### ISSUE THREAD

**Issue by Coeur-Noir** [libredeb/lightpad#11]
Hi,

Lightpad is a lovely looking app-menu. But DE don't really lack of app-menu.
Whereas they almost all lack of a one-place, easy to reach, access to many recent activities.

What do I mean here by activities ?
⋅ recently used files,
⋅ recently visited folders,
⋅ recently used app's ( ideally only those not already pinned into a dock or a panel )

Those recent/frequent things may be followed through _tracker_.

On the same layout I'd see something like
⋅ 2 first lines for files,
⋅ 3rd line for folders
⋅ last line for app's
⋅ first page for the most recent, last page for the oldest

User should have the ability to
⋅ set the default view of lightpad on _app-menu_ or _recent-activities_
⋅ switch between those two types of view ( click on an icon or keyboard shortcut ),
⋅ set the span time for recent activities ( last _x_ hours or days with a reasonable maximum, 23h/31d )
⋅ purge that recent-activity view ( = reset time counters ↑ to zero ),
⋅ set a blacklist of app's not to show in recent view if this can't be retrieved automatically from docks or panels-applets.

Well, that sounds like a huge amount of work. But it would give lightpad a set of very distinctive feature none others offer ;-)

**Comment by libredeb** [comment]
@Coeur-Noir yes, maybe can use the file $HOME/.local/share/recently-used.xbel (in the case of GNOME for example)... this file maintain recently used files. But we will have to evaluate and test to contemplate the best way to implement it and review how many resources it consumes since the main idea is not only to have a light and powerful alternative to the macOS launchpad, but also simple. Big hug!

**Comment by Coeur-Noir** [comment]
Well, I did not at all think about « light » here, but « useful » or « distinctive » when compared to all other app-menu out there.

Glad you answered ;-)

**Comment by libredeb** [comment]
@Coeur-Noir Hi,

Yes I understand the point... but the original idea behind LightPad is to offer an alternative to what exists in the macOS world with launchpad and I don't want it to lose its essence. I hope you understand.

Anyway we do not rule out including it in the future as an optional configuration.

Regards!
